test-data/unit/check-modules.test

```diff
@@ -1943,3 +1943,24 @@ main:2: error: Name 'name' is not defined
 main:3: error: Revealed type is 'Any'
 
 [builtins fixtures/module.pyi]
+
+
+[case testImportCycle]
+from m import One
+reveal_type(One)
+[file m/__init__.py]
+from .one import One
+from .two import Two
+reveal_type(One)
+[file m/one.py]
+class One:
+    pass
+[file m/two.py]
+from m import One
+reveal_type(One)
+class Two:
+    pass
+[out]
```

Comment:
Please also add tests for type aliases (you seem to care about them in `visit_import_from`).

Reply:
So the type-alias stuff is there just because it seemed better to be complete about transferring node info (in the same way that `visit_import_from` in second pass is complete), rather than only transfer some of the node attributes. I have not had any real-world issue with type aliases and circular imports that those lines are intended to solve. I will try to see if I can repro some case involving type aliases that is fixed by those lines.